scipy/optimize/tests/test_isotonic_regression.py

```diff
@@ -16,7 +16,9 @@ class TestIsotonicRegression:
              "Input arrays y and w must have one dimension of equal length"),
             ([0, 1], [1],
              "Input arrays y and w must have one dimension of equal length"),
-            (1, 2,
+            (1, [1, 2],
+             "Input arrays y and w must have one dimension of equal length"),
+            ([1, 2], 1,
```

Comment:
Why do we need to change the regression tests in a PR that only aims to avoid extra array copies? If Evgeni is happy with this I don't mind, but why do we need to get rid of the old test case?

Reply:
Because `isotonic_regression(y=1, weights=2)` should work. In current main branch, the scalar values are not promoted to arrays of shape `(1,)` and an error is raised.
With this PR they are reshaped and no error is raised anymore.
Passing scalar values is an edge case because the solution is trivial (return the same values), and no proper usage of isotonic regression will/should do that.